html/captive-portal/lib/captiveportal/PacketFence/Controller/Status.pm

```diff
@@ -76,13 +76,16 @@ sub is_lost_stolen {
 sub userIsAuthenticated : Private {
     my ( $self, $c ) = @_;
     my $pid   = $c->user_session->{"username"};
-    my @nodes = person_nodes($pid);
-    foreach my $node (@nodes) {
+    my @person_nodes = person_nodes($pid);
```

Comment:
To avoid doing two database calls.
Add the additional fields and joins [here](https://github.com/inverse-inc/packetfence/blob/devel/lib/pf/person.pm#L264) instead.

Reply:
From what I recall that specific join was removed from this query (even before DAL) because it was causing latency at some 3 letter telecom company which has a very large node table.

So that is why I didn't add it there since this is called in the RADIUS to check the max nodes

Reply:
person_nodes is currently only used in the following
```
html/captive-portal/lib/captiveportal/PacketFence/Controller/Status.pm
html/pfappserver/lib/pfappserver/PacketFence/Model/User.pm
```
Unless my grepping is off I do not see how where is this called in RADIUS.

As for what I can tell we use pf::node::node_pid to check max nodes.

Reply:
I'll change this in that case, my memory has failed me

I shall blame @lzammit anyway